<a href="https://colab.research.google.com/github/Kaavyesh/Customer_acquisition_cost_casestudy/blob/main/customer_acquisition_cost_casestudy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Stage 1: Data Preprocessing

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
import numpy as np
import plotly.express as px

In [3]:
path = '/content/drive/MyDrive/custome acquisition cost case study /customer_acquisition_cost_dataset.csv'

In [4]:
df = pd.read_csv(path)

In [5]:
df.head()

Customer_ID Marketing_Channel  Marketing_Spend  New_Customers
0    CUST0001   Email Marketing      3489.027844             16
1    CUST0002        Online Ads      1107.865808             33
2    CUST0003      Social Media      2576.081025             44
3    CUST0004        Online Ads      3257.567932             32
4    CUST0005   Email Marketing      1108.408185             13

In [6]:
df.shape

(500, 4)

In [7]:
df.isnull().sum()

Customer_ID          0
Marketing_Channel    0
Marketing_Spend      0
New_Customers        0
dtype: int64

In [8]:
df.duplicated().sum()

0

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Customer_ID        500 non-null    object 
 1   Marketing_Channel  500 non-null    object 
 2   Marketing_Spend    500 non-null    float64
 3   New_Customers      500 non-null    int64  
dtypes: float64(1), int64(1), object(2)
memory usage: 15.8+ KB


In [10]:
df.describe()


Marketing_Spend  New_Customers
count       500.000000     500.000000
mean       3094.726549      30.070000
std        1136.628605      11.754209
min        1010.702890      10.000000
25%        2131.835093      19.000000
50%        3155.976673      31.000000
75%        4027.494116      40.000000
max        4999.631314      50.000000

In [11]:
df.columns

Index(['Customer_ID', 'Marketing_Channel', 'Marketing_Spend', 'New_Customers'], dtype='object')

##Stage 2 : Caluculate cac

In [13]:
channel_groups = df.groupby('Marketing_Channel').agg({'Marketing_Spend':'sum', 'New_Customers':'sum'})

In [14]:
channel_groups

Marketing_Spend  New_Customers
Marketing_Channel                                
Email Marketing      384034.640088           3583
Online Ads           388747.870215           3896
Referral             391420.513169           3904
Social Media         383160.250917           3652

In [16]:
channel_groups['CAC'] = channel_groups['Marketing_Spend'] / channel_groups['New_Customers']

In [17]:
channel_groups

Marketing_Spend  New_Customers         CAC
Marketing_Channel                                            
Email Marketing      384034.640088           3583  107.182428
Online Ads           388747.870215           3896   99.781281
Referral             391420.513169           3904  100.261402
Social Media         383160.250917           3652  104.917922

In [22]:
channel_groups = channel_groups.sort_values(by='CAC')

In [23]:
channel_groups

Marketing_Spend  New_Customers         CAC
Marketing_Channel                                            
Online Ads           388747.870215           3896   99.781281
Referral             391420.513169           3904  100.261402
Social Media         383160.250917           3652  104.917922
Email Marketing      384034.640088           3583  107.182428

In [24]:
df.columns

Index(['Customer_ID', 'Marketing_Channel', 'Marketing_Spend', 'New_Customers'], dtype='object')

##Step 3: Data Visualization

In [36]:
fig = px.bar(channel_groups,x = channel_groups.index , y = 'CAC' ,hover_name='CAC',title = 'CAC by Marketing Channel',color = channel_groups.index )
fig.show()

In [37]:
fig = px.bar(channel_groups, x=channel_groups.index,
             y=['CAC', 'New_Customers'],
             title='Comparison of CAC and New Customers by Marketing Channel',
             labels={'CAC': 'Customer Acquisition Cost (CAC)', 'New_Customers': 'New Customers'},
             color_discrete_map={'CAC': 'blue', 'New_Customers': 'green'},
             barmode='group')  # barmode='group' places bars side by side

# Show the chart
fig.show()

In [40]:
fig_pie = px.pie(channel_groups,names = channel_groups.index , values = 'CAC',color = channel_groups.index,hover_name = channel_groups.index,hover_data = 'CAC',title = 'CAC Distribution by Marketing Channel')
fig_pie.show()

##Step 4: Insights and Recommendations

###**Comparison of CAC Across Channels:**

Online Ads have the lowest CAC at approximately $99.78 per customer acquired.

Referral marketing is close behind, with a CAC of about $100.26 per customer.

Social Media has a slightly higher CAC of around $104.92 per customer.

Email Marketing has the highest CAC among the channels, at approximately $107.18 per customer.

###**Marketing Efficiency:**

Online Ads and Referral marketing seem to be the most cost-effective channels in terms of acquiring new customers.

Social Media is moderately cost-effective, while Email Marketing appears to be the least efficient in terms of cost-effectiveness.